Stacktrace from /mail delete

Running `/mail delete` with no arguments causes a stacktrace.

org.bukkit.command.CommandException: Unhandled exception executing command 'mail' in plugin NerdMessage v1.3.1
?at org.bukkit.command.PluginCommand.execute(PluginCommand.java:47) ~[patched_1.16.5.jar:git-Paper-778]
?at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:159) ~[patched_1.16.5.jar:git-Paper-778]
?at org.bukkit.craftbukkit.v1_16_R3.CraftServer.dispatchCommand(CraftServer.java:826) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.PlayerConnection.handleCommand(PlayerConnection.java:2185) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.PlayerConnection.c(PlayerConnection.java:2000) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.PlayerConnection.a(PlayerConnection.java:1953) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.PacketPlayInChat.a(PacketPlayInChat.java:49) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.PacketPlayInChat.a(PacketPlayInChat.java:7) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.PlayerConnectionUtils.lambda$ensureMainThread$1(PlayerConnectionUtils.java:35) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.TickTask.run(SourceFile:18) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.MinecraftServer.bb(MinecraftServer.java:1266) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.MinecraftServer.executeNext(MinecraftServer.java:1259) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeAll(IAsyncTaskHandler.java:95) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1395) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1130) ~[patched_1.16.5.jar:git-Paper-778]
?at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:291) ~[patched_1.16.5.jar:git-Paper-778]
?at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
?at nu.nerd.nerdmessage.commands.MailCommands.archiveCommand(MailCommands.java:280) ~[?:?]
?at nu.nerd.nerdmessage.commands.MailCommands.onCommand(MailCommands.java:67) ~[?:?]
?at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45) ~[patched_1.16.5.jar:git-Paper-778]

--- a/pom.xml
+++ b/pom.xml
@@ -3,7 +3,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>nu.nerd</groupId>
     <artifactId>nerdmessage</artifactId>
-    <version>1.3.3</version>
+    <version>1.3.4</version>
     <packaging>jar</packaging>
     <name>NerdMessage</name>
     <description>Messaging plugin for Bukkit</description>

--- a/src/nu/nerd/nerdmessage/commands/MailCommands.java
+++ b/src/nu/nerd/nerdmessage/commands/MailCommands.java
@@ -226,6 +226,10 @@ public class MailCommands implements CommandExecutor {
             public void run() {
 
                 List<MailMessage> messages = MailMessage.findUnread(player.getUniqueId());
+                for(MailMessage mail : messages) {
+                    mail.setNotified(true);
+                    MailMessage.flagNotified(player.getUniqueId());
+                }
                 int pages = (messages.size() + perPage - 1) / perPage; //integer division
                 int offset = (page - 1) * perPage;
 

--- a/src/nu/nerd/nerdmessage/commands/MailCommands.java
+++ b/src/nu/nerd/nerdmessage/commands/MailCommands.java
@@ -281,7 +281,7 @@ public class MailCommands implements CommandExecutor {
 
         Integer id;
 
-        if (args[1].equals("")) {
+        if(args.length == 1) {
             sender.sendMessage(ChatColor.RED + "Usage: /mail archive <id>");
             return;
         }

--- a/src/nu/nerd/nerdmessage/mail/MailHandler.java
+++ b/src/nu/nerd/nerdmessage/mail/MailHandler.java
@@ -97,7 +97,6 @@ public class MailHandler implements Listener {
             public void run() {
                 if (MailMessage.findUnnotified(player.getUniqueId()).size() > 0) {
                     notifyNewMessages(player.getUniqueId(), true);
-                    MailMessage.flagNotified(player.getUniqueId());
                 }
             }
         }.runTaskLaterAsynchronously(plugin, 40L);
@@ -112,21 +111,14 @@ public class MailHandler implements Listener {
      * @param isOrigin this is the server the message is originating from
      */
     public void notifyNewMessages(UUID recipient, boolean isOrigin) {
-        boolean notified = false;
         Player player = plugin.getServer().getPlayer(recipient);
         String msg = String.format("%sYou have new mail! Type %s/mail inbox%s to read it.", ChatColor.GREEN, ChatColor.LIGHT_PURPLE, ChatColor.GREEN);
         if (player != null && player.isOnline()) {
             player.sendMessage(msg);
-            notified = true;
         } else {
             if (plugin.crossServerEnabled() && isOrigin) {
                 plugin.redisPublish("mail.new", recipient.toString());
             }
         }
-        if (notified) {
-            MailMessage.flagNotified(recipient);
-        }
     }
-
-
 }

